include/ozo/type_traits.h

```diff
@@ -92,11 +113,53 @@ inline auto is_null(const ::std::weak_ptr<T>& v) noexcept {
 }
 
 template <typename T>
-constexpr auto is_null(const T&) noexcept ->
-    std::enable_if_t<!is_nullable<std::decay_t<T>>::value, std::false_type> {
+constexpr auto is_null(const T&) noexcept -> std::enable_if_t<!Nullable<T>, std::false_type> {
     return {};
 }
 
+template <typename T>
+void init_nullable(T&) {}
+
+template <typename T>
+void init_nullable(::boost::optional<T>& o) {
+    o.emplace();
+}
+
+#ifdef __OZO_STD_OPTIONAL
+template <typename T>
+void init_nullable(__OZO_STD_OPTIONAL<T>& o) {
+    o.emplace();
+}
+#endif
+
+template <typename T>
+void init_nullable(::boost::scoped_ptr<T>& ptr) {
```

Comment:
IMO for init_nullable allocator must be provided. Maybe with std::allocator as default.